puppet/custom_device_context.py

```diff
@@ -1,61 +1,83 @@
-    for record in data:
-        for _, device in record.items():
-            device_details = device["facts"]
-            api_payload = {
-                "mac_address": device_details["macaddress"],
-                "ip": device_details["ipaddress"],
-                "source": SOURCE,
-                "hostname": device_details["hostname"],
-                "manufacturer": device_details["manufacturer"],
-                "os": device_details.get("os", {}).get("name"),
-                "username": device_details.get("identity", {}).get("user"),
-                "use_asset": True
-            }
-            response = requests.post("https://api.wootuno.wootcloud.com/v1/integrations/custom_devicecontext",
-                                     auth=HTTPBasicAuth(client_id, secret_key),
-                                     json=api_payload)
-            if response.status_code == 202:
-                print(f"Accepted")
-            elif response.status_code in [401, 429]:
-                print(f"Failed with reason: {response.reason}")
-            elif response.status_code == 400:
-                response = response.json()
-                print(f"Failed with reason: {response['message']}. Payload: {api_payload}")
+@dataclass
+class PuppetInterface:
+    mac_address: str
+    ip: str
+    hostname: str
+    manufacturer: str
+    os: str
+    username: str
+
+    @classmethod
+    def from_json(cls, data):
+        return cls(
+            mac_address=data.get("macaddress"),
+            ip=data.get("ipaddress"),
+            hostname=data.get("hostname"),
+            manufacturer=data.get("manufacturer"),
+            os=data.get("os", {}).get("name"),
+            username=data.get("identity", {}).get("user")
+        )
+
+
+def read_data_file(file: io.TextIOWrapper) -> List[PuppetInterface]:
+    interfaces = []
+    with file as f:
+        for record in json.load(f):
+            for _, interface in record.items():
+                interfaces.append(PuppetInterface.from_json(interface["facts"]))
+    return interfaces
+
+
+# Maps puppet data to payload and calls devicecontext API
+def map_and_call_device_context_api(puppet_interfaces: List[PuppetInterface], client_id: str, secret_key: str):
+    for puppet_interface in puppet_interfaces:
+        api_payload = {
+            "mac_address": puppet_interface.mac_address,
+            "ip": puppet_interface.ip,
+            "source": SOURCE,
+            "hostname": puppet_interface.hostname,
+            "manufacturer": puppet_interface.manufacturer,
+            "os": puppet_interface.os,
+            "username": puppet_interface.username,
+            "use_asset": True
+        }
+        response = requests.post(API_URI, auth=HTTPBasicAuth(client_id, secret_key), json=api_payload)
+        if response.status_code == 202:
+            print(f"Accepted")
+        elif response.status_code in [401, 429]:
+            print(f"Failed with reason: {response.reason}")
```

Comment:
Same, include the mac address in the message.

Reply:
Fixed